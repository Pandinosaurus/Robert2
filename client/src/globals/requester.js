import axios from 'axios';
import deepObjectSet from 'lodash/set';
import isPlainObject from 'lodash/isPlainObject';
import flattenObject from '@/utils/flattenObject';
import config from '@/globals/config';
import cookies from '@/utils/cookies';
import DateTime from '@/utils/datetime';
import Day from '@/utils/day';

const requester = axios.create({
    baseURL: config.api.url,
    headers: config.api.headers,
});

requester.interceptors.request.use(
    (_request) => {
        const { params, onProgress, ...request } = _request;

        // - Ajoute une méthode haut-niveau permettant de récupérer l'avancement sous forme numérique.
        //   - Si la méthode de la requête est POST, PUT ou PATCH, c'est l'événement `onUploadProgress` qui sera écouté.
        //   - Sinon ce sera `onDownloadProgress`
        if (onProgress) {
            const progressCallback = (event) => {
                if (!event.lengthComputable) {
                    return;
                }

                const { loaded, total } = event;
                const percent = (loaded / total) * 100;
                onProgress(percent);
            };

            if (request.method && ['POST', 'PUT', 'PATCH'].includes(request.method.toUpperCase())) {
                request.onUploadProgress = progressCallback;
            } else {
                request.onDownloadProgress = progressCallback;
            }
        }

        // - Traitement spécial des objets littéraux contenant des fichiers.
        //   => On utilise un envoi via `multipart/form-data` et on met les données
        //      supplémentaires éventuelles dans une clé spéciale `@data`)
        const hasFiles = (
            isPlainObject(request.data) &&
            Object.values(request.data).some((value) => (
                value instanceof File
            ))
        );
        if (hasFiles) {
            const formData = new FormData();
            const data = Object.entries(flattenObject(request.data)).reduce(
                (_data, [key, value]) => {
                    if (value instanceof File) {
                        formData.append(btoa(key), value);
                        return _data;
                    }
                    return deepObjectSet(_data, key, value);
                },
                {},
            );

            formData.append('@data', JSON.stringify(data));
            request.data = formData;
        }

        if (params) {
            Object.keys(params).forEach((name) => {
                if (params[name] === true) {
                    params[name] = '1';
                }
                if (params[name] === false) {
                    params[name] = '0';
                }
                if (params[name] instanceof DateTime || params[name] instanceof Day) {
                    params[name] = params[name].toString();
                }
            });
            request.params = params;
        }

        const authToken = cookies.get(config.auth.cookie);
        if (authToken) {
            request.headers.Authorization = `Bearer ${authToken}`;
        }
        return request;
    },
    (error) => Promise.reject(error),
);

export default requester;
