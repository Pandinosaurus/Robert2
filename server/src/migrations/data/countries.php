<?php
declare(strict_types=1);

return [
    // ------------------------------------------------------
    // -
    // -    Francophones
    // -
    // ------------------------------------------------------

    ['code' => 'FR', 'name' => 'France'],
    ['code' => 'BE', 'name' => 'Belgique'],
    ['code' => 'CH', 'name' => 'Suisse'],
    ['code' => 'CA', 'name' => 'Canada'],
    ['code' => 'MC', 'name' => 'Monaco'],
    ['code' => 'LU', 'name' => 'Luxembourg'],

    ['code' => 'GP', 'name' => 'Guadeloupe'],
    ['code' => 'GF', 'name' => 'Guyane française'],
    ['code' => 'MQ', 'name' => 'Martinique'],
    ['code' => 'YT', 'name' => 'Mayotte'],
    ['code' => 'NC', 'name' => 'Nouvelle-Calédonie'],
    ['code' => 'PF', 'name' => 'Polynésie française'],
    ['code' => 'RE', 'name' => 'Réunion'],
    ['code' => 'BL', 'name' => 'Saint-Barthélemy'],
    ['code' => 'MF', 'name' => 'Saint-Martin'],
    ['code' => 'PM', 'name' => 'Saint-Pierre-et-Miquelon'],
    ['code' => 'TF', 'name' => 'Terres australes françaises'],
    ['code' => 'WF', 'name' => 'Wallis-et-Futuna'],

    // ------------------------------------------------------
    // -
    // -    Non francophones
    // -
    // ------------------------------------------------------

    ['code' => 'AF', 'name' => "Afghanistan"],
    ['code' => 'ZA', 'name' => "Afrique du Sud"],
    ['code' => 'AL', 'name' => "Albanie"],
    ['code' => 'DZ', 'name' => "Algérie"],
    ['code' => 'DE', 'name' => "Allemagne"],
    ['code' => 'AD', 'name' => "Andorre"],
    ['code' => 'AO', 'name' => "Angola"],
    ['code' => 'AI', 'name' => "Anguilla"],
    ['code' => 'AQ', 'name' => "Antarctique"],
    ['code' => 'AG', 'name' => "Antigua-et-Barbuda"],
    ['code' => 'AN', 'name' => "Antilles néerlandaises"],
    ['code' => 'SA', 'name' => "Arabie saoudite"],
    ['code' => 'AR', 'name' => "Argentine"],
    ['code' => 'AM', 'name' => "Arménie"],
    ['code' => 'AW', 'name' => "Aruba"],
    ['code' => 'AU', 'name' => "Australie"],
    ['code' => 'AT', 'name' => "Autriche"],
    ['code' => 'AZ', 'name' => "Azerbaïdjan"],
    ['code' => 'BS', 'name' => "Bahamas"],
    ['code' => 'BH', 'name' => "Bahreïn"],
    ['code' => 'BD', 'name' => "Bangladesh"],
    ['code' => 'BB', 'name' => "Barbade"],
    ['code' => 'BZ', 'name' => "Belize"],
    ['code' => 'BJ', 'name' => "Bénin"],
    ['code' => 'BM', 'name' => "Bermudes"],
    ['code' => 'BT', 'name' => "Bhoutan"],
    ['code' => 'BY', 'name' => "Biélorussie"],
    ['code' => 'MM', 'name' => "Birmanie (Myanmar)"],
    ['code' => 'BO', 'name' => "Bolivie"],
    ['code' => 'BA', 'name' => "Bosnie-Herzégovine"],
    ['code' => 'BW', 'name' => "Botswana"],
    ['code' => 'BR', 'name' => "Brésil"],
    ['code' => 'BN', 'name' => "Brunéi Darussalam"],
    ['code' => 'BG', 'name' => "Bulgarie"],
    ['code' => 'BF', 'name' => "Burkina Faso"],
    ['code' => 'BI', 'name' => "Burundi"],
    ['code' => 'KH', 'name' => "Cambodge"],
    ['code' => 'CM', 'name' => "Cameroun"],
    ['code' => 'CV', 'name' => "Cap-Vert"],
    ['code' => 'CL', 'name' => "Chili"],
    ['code' => 'CN', 'name' => "Chine"],
    ['code' => 'CY', 'name' => "Chypre"],
    ['code' => 'VA', 'name' => "Cité du Vatican"],
    ['code' => 'CO', 'name' => "Colombie"],
    ['code' => 'KM', 'name' => "Comores"],
    ['code' => 'CG', 'name' => "Congo"],
    ['code' => 'KP', 'name' => "Corée du Nord"],
    ['code' => 'KR', 'name' => "Corée du Sud"],
    ['code' => 'CR', 'name' => "Costa Rica"],
    ['code' => 'CI', 'name' => "Côte d'Ivoire"],
    ['code' => 'HR', 'name' => "Croatie"],
    ['code' => 'CU', 'name' => "Cuba"],
    ['code' => 'DK', 'name' => "Danemark"],
    ['code' => 'DJ', 'name' => "Djibouti"],
    ['code' => 'DM', 'name' => "Dominique"],
    ['code' => 'EG', 'name' => "Égypte"],
    ['code' => 'AE', 'name' => "Émirats arabes unis"],
    ['code' => 'EC', 'name' => "Équateur"],
    ['code' => 'ER', 'name' => "Érythrée"],
    ['code' => 'ES', 'name' => "Espagne"],
    ['code' => 'EE', 'name' => "Estonie"],
    ['code' => 'SZ', 'name' => "Eswatini"],
    ['code' => 'PS', 'name' => "État de Palestine"],
    ['code' => 'US', 'name' => "États-Unis"],
    ['code' => 'ET', 'name' => "Éthiopie"],
    ['code' => 'FJ', 'name' => "Fidji"],
    ['code' => 'FI', 'name' => "Finlande"],
    ['code' => 'GA', 'name' => "Gabon"],
    ['code' => 'GM', 'name' => "Gambie"],
    ['code' => 'GE', 'name' => "Géorgie"],
    ['code' => 'GS', 'name' => "Géorgie du Sud-et-les Îles Sandwich du Sud"],
    ['code' => 'GH', 'name' => "Ghana"],
    ['code' => 'GI', 'name' => "Gibraltar"],
    ['code' => 'GR', 'name' => "Grèce"],
    ['code' => 'GD', 'name' => "Grenade"],
    ['code' => 'GL', 'name' => "Groenland"],
    ['code' => 'GU', 'name' => "Guam"],
    ['code' => 'GT', 'name' => "Guatemala"],
    ['code' => 'GG', 'name' => "Guernesey"],
    ['code' => 'GN', 'name' => "Guinée"],
    ['code' => 'GQ', 'name' => "Guinée équatoriale"],
    ['code' => 'GW', 'name' => "Guinée-Bissau"],
    ['code' => 'GY', 'name' => "Guyana"],
    ['code' => 'HT', 'name' => "Haïti"],
    ['code' => 'HN', 'name' => "Honduras"],
    ['code' => 'HK', 'name' => "Hong Kong"],
    ['code' => 'HU', 'name' => "Hongrie"],
    ['code' => 'BV', 'name' => "Île Bouvet"],
    ['code' => 'CX', 'name' => "Île Christmas"],
    ['code' => 'IM', 'name' => "Île de Man"],
    ['code' => 'NF', 'name' => "Île Norfolk"],
    ['code' => 'AX', 'name' => "Îles Åland"],
    ['code' => 'KY', 'name' => "Îles Caïmans"],
    ['code' => 'CC', 'name' => "Îles Cocos (Keeling)"],
    ['code' => 'CK', 'name' => "Îles Cook"],
    ['code' => 'FK', 'name' => "Îles Falkland (Malouines)"],
    ['code' => 'FO', 'name' => "Îles Féroé"],
    ['code' => 'HM', 'name' => "Îles Heard-et-MacDonald"],
    ['code' => 'MP', 'name' => "Îles Mariannes du Nord"],
    ['code' => 'MH', 'name' => "Îles Marshall"],
    ['code' => 'SB', 'name' => "Îles Salomon"],
    ['code' => 'TC', 'name' => "Îles Turques-et-Caïques"],
    ['code' => 'IN', 'name' => "Inde"],
    ['code' => 'ID', 'name' => "Indonésie"],
    ['code' => 'IQ', 'name' => "Irak"],
    ['code' => 'IR', 'name' => "Iran"],
    ['code' => 'IE', 'name' => "Irlande"],
    ['code' => 'IS', 'name' => "Islande"],
    ['code' => 'IL', 'name' => "Israël"],
    ['code' => 'IT', 'name' => "Italie"],
    ['code' => 'JM', 'name' => "Jamaïque"],
    ['code' => 'JP', 'name' => "Japon"],
    ['code' => 'JE', 'name' => "Jersey"],
    ['code' => 'JO', 'name' => "Jordanie"],
    ['code' => 'KZ', 'name' => "Kazakhstan"],
    ['code' => 'KE', 'name' => "Kenya"],
    ['code' => 'KG', 'name' => "Kirghizistan"],
    ['code' => 'KI', 'name' => "Kiribati"],
    ['code' => 'KW', 'name' => "Koweït"],
    ['code' => 'LA', 'name' => "Laos"],
    ['code' => 'LS', 'name' => "Lesotho"],
    ['code' => 'LV', 'name' => "Lettonie"],
    ['code' => 'LB', 'name' => "Liban"],
    ['code' => 'LR', 'name' => "Libéria"],
    ['code' => 'LY', 'name' => "Libye"],
    ['code' => 'LI', 'name' => "Liechtenstein"],
    ['code' => 'LT', 'name' => "Lituanie"],
    ['code' => 'MO', 'name' => "Macao"],
    ['code' => 'MK', 'name' => "Macédoine du Nord"],
    ['code' => 'MG', 'name' => "Madagascar"],
    ['code' => 'MY', 'name' => "Malaisie"],
    ['code' => 'MW', 'name' => "Malawi"],
    ['code' => 'MV', 'name' => "Maldives"],
    ['code' => 'ML', 'name' => "Mali"],
    ['code' => 'MT', 'name' => "Malte"],
    ['code' => 'MA', 'name' => "Maroc"],
    ['code' => 'MU', 'name' => "Maurice"],
    ['code' => 'MR', 'name' => "Mauritanie"],
    ['code' => 'MX', 'name' => "Mexique"],
    ['code' => 'FM', 'name' => "Micronésie"],
    ['code' => 'MD', 'name' => "Moldavie"],
    ['code' => 'MN', 'name' => "Mongolie"],
    ['code' => 'ME', 'name' => "Monténégro"],
    ['code' => 'MS', 'name' => "Montserrat"],
    ['code' => 'MZ', 'name' => "Mozambique"],
    ['code' => 'NA', 'name' => "Namibie"],
    ['code' => 'NR', 'name' => "Nauru"],
    ['code' => 'NP', 'name' => "Népal"],
    ['code' => 'NI', 'name' => "Nicaragua"],
    ['code' => 'NE', 'name' => "Niger"],
    ['code' => 'NG', 'name' => "Nigeria"],
    ['code' => 'NU', 'name' => "Niue"],
    ['code' => 'NO', 'name' => "Norvège"],
    ['code' => 'NZ', 'name' => "Nouvelle-Zélande"],
    ['code' => 'OM', 'name' => "Oman"],
    ['code' => 'UG', 'name' => "Ouganda"],
    ['code' => 'UZ', 'name' => "Ouzbékistan"],
    ['code' => 'PK', 'name' => "Pakistan"],
    ['code' => 'PW', 'name' => "Palaos"],
    ['code' => 'PA', 'name' => "Panama"],
    ['code' => 'PG', 'name' => "Papouasie-Nouvelle-Guinée"],
    ['code' => 'PY', 'name' => "Paraguay"],
    ['code' => 'NL', 'name' => "Pays-Bas"],
    ['code' => 'PE', 'name' => "Pérou"],
    ['code' => 'PH', 'name' => "Philippines"],
    ['code' => 'PN', 'name' => "Pitcairn"],
    ['code' => 'PL', 'name' => "Pologne"],
    ['code' => 'PR', 'name' => "Porto Rico"],
    ['code' => 'PT', 'name' => "Portugal"],
    ['code' => 'QA', 'name' => "Qatar"],
    ['code' => 'CF', 'name' => "République centrafricaine"],
    ['code' => 'CD', 'name' => "République démocratique du Congo"],
    ['code' => 'DO', 'name' => "République dominicaine"],
    ['code' => 'CZ', 'name' => "République tchèque"],
    ['code' => 'RO', 'name' => "Roumanie"],
    ['code' => 'GB', 'name' => "Royaume-Uni"],
    ['code' => 'RU', 'name' => "Russie"],
    ['code' => 'RW', 'name' => "Rwanda"],
    ['code' => 'EH', 'name' => "Sahara occidental"],
    ['code' => 'KN', 'name' => "Saint-Kitts-et-Nevis"],
    ['code' => 'SM', 'name' => "Saint-Marin"],
    ['code' => 'VC', 'name' => "Saint-Vincent-et-les-Grenadines"],
    ['code' => 'SH', 'name' => "Sainte-Hélène"],
    ['code' => 'LC', 'name' => "Sainte-Lucie"],
    ['code' => 'SV', 'name' => "Salvador"],
    ['code' => 'WS', 'name' => "Samoa"],
    ['code' => 'AS', 'name' => "Samoa américaines"],
    ['code' => 'ST', 'name' => "Sao Tomé-et-Principe"],
    ['code' => 'SN', 'name' => "Sénégal"],
    ['code' => 'RS', 'name' => "Serbie"],
    ['code' => 'SC', 'name' => "Seychelles"],
    ['code' => 'SL', 'name' => "Sierra Leone"],
    ['code' => 'SG', 'name' => "Singapour"],
    ['code' => 'SK', 'name' => "Slovaquie"],
    ['code' => 'SI', 'name' => "Slovénie"],
    ['code' => 'SO', 'name' => "Somalie"],
    ['code' => 'SD', 'name' => "Soudan"],
    ['code' => 'LK', 'name' => "Sri Lanka"],
    ['code' => 'SE', 'name' => "Suède"],
    ['code' => 'SR', 'name' => "Suriname"],
    ['code' => 'SJ', 'name' => "Svalbard et Jan Mayen"],
    ['code' => 'SY', 'name' => "Syrie"],
    ['code' => 'TJ', 'name' => "Tadjikistan"],
    ['code' => 'TW', 'name' => "Taïwan"],
    ['code' => 'TZ', 'name' => "Tanzanie"],
    ['code' => 'TD', 'name' => "Tchad"],
    ['code' => 'IO', 'name' => "Territoire britannique de l'océan Indien"],
    ['code' => 'TH', 'name' => "Thaïlande"],
    ['code' => 'TL', 'name' => "Timor-Leste"],
    ['code' => 'TG', 'name' => "Togo"],
    ['code' => 'TK', 'name' => "Tokelau"],
    ['code' => 'TO', 'name' => "Tonga"],
    ['code' => 'TT', 'name' => "Trinité-et-Tobago"],
    ['code' => 'TN', 'name' => "Tunisie"],
    ['code' => 'TM', 'name' => "Turkménistan"],
    ['code' => 'TR', 'name' => "Turquie"],
    ['code' => 'TV', 'name' => "Tuvalu"],
    ['code' => 'UA', 'name' => "Ukraine"],
    ['code' => 'UY', 'name' => "Uruguay"],
    ['code' => 'VU', 'name' => "Vanuatu"],
    ['code' => 'VE', 'name' => "Venezuela"],
    ['code' => 'VN', 'name' => "Viêt Nam"],
    ['code' => 'YE', 'name' => "Yémen"],
    ['code' => 'ZM', 'name' => "Zambie"],
    ['code' => 'ZW', 'name' => "Zimbabwe"],
];
