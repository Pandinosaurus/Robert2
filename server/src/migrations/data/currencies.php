<?php
declare(strict_types=1);

return [
    // ------------------------------------------------------
    // -
    // -    Francophones
    // -
    // ------------------------------------------------------

    ['code' => 'EUR', 'name' => 'Euro'],
    ['code' => 'CHF', 'name' => 'Franc suisse'],
    ['code' => 'CAD', 'name' => 'Dollar canadien'],
    ['code' => 'BIF', 'name' => 'Franc burundais'],
    ['code' => 'CDF', 'name' => 'Franc congolais'],
    ['code' => 'DJF', 'name' => 'Franc djiboutien'],
    ['code' => 'DZD', 'name' => 'Dinar algérien'],
    ['code' => 'GNF', 'name' => 'Franc guinéen'],
    ['code' => 'KMF', 'name' => 'Franc comorien'],
    ['code' => 'MAD', 'name' => 'Dirham marocain'],
    ['code' => 'RWF', 'name' => 'Franc rwandais'],
    ['code' => 'TND', 'name' => 'Dinar tunisien'],
    ['code' => 'XAF', 'name' => 'Franc CFA (BEAC)'],
    ['code' => 'XOF', 'name' => 'Franc CFA (BCEAO)'],
    ['code' => 'XPF', 'name' => 'Franc pacifique'],

    // ------------------------------------------------------
    // -
    // -    Non francophones
    // -
    // ------------------------------------------------------

    ['code' => 'AFN', 'name' => "Afghani afghan"],
    ['code' => 'MGA', 'name' => "Ariary malgache"],
    ['code' => 'THB', 'name' => "Baht thaïlandais"],
    ['code' => 'PAB', 'name' => "Balboa panaméen"],
    ['code' => 'ETB', 'name' => "Birr éthiopien"],
    ['code' => 'VES', 'name' => "Bolívar vénézuélien"],
    ['code' => 'BOB', 'name' => "Boliviano bolivien"],
    ['code' => 'GHS', 'name' => "Cedi ghanéen"],
    ['code' => 'CRC', 'name' => "Colón costaricien"],
    ['code' => 'NIO', 'name' => "Córdoba nicaraguayen"],
    ['code' => 'DKK', 'name' => "Couronne danoise"],
    ['code' => 'NOK', 'name' => "Couronne norvégienne"],
    ['code' => 'SEK', 'name' => "Couronne suédoise"],
    ['code' => 'CZK', 'name' => "Couronne tchèque"],
    ['code' => 'BHD', 'name' => "Dinar bahreïni"],
    ['code' => 'IQD', 'name' => "Dinar irakien"],
    ['code' => 'JOD', 'name' => "Dinar jordanien"],
    ['code' => 'KWD', 'name' => "Dinar koweïtien"],
    ['code' => 'LYD', 'name' => "Dinar libyen"],
    ['code' => 'MKD', 'name' => "Dinar macédonien"],
    ['code' => 'RSD', 'name' => "Dinar serbe"],
    ['code' => 'AED', 'name' => "Dirham des Émirats arabes unis"],
    ['code' => 'USD', 'name' => "Dollar américain"],
    ['code' => 'AUD', 'name' => "Dollar australien"],
    ['code' => 'BZD', 'name' => "Dollar bélizien"],
    ['code' => 'BND', 'name' => "Dollar brunéien"],
    ['code' => 'HKD', 'name' => "Dollar de Hong Kong"],
    ['code' => 'SGD', 'name' => "Dollar de Singapour"],
    ['code' => 'TWD', 'name' => "Dollar de Taïwan"],
    ['code' => 'TTD', 'name' => "Dollar de Trinité-et-Tobago"],
    ['code' => 'JMD', 'name' => "Dollar jamaïcain"],
    ['code' => 'NAD', 'name' => "Dollar namibien"],
    ['code' => 'NZD', 'name' => "Dollar néo-zélandais"],
    ['code' => 'ZWL', 'name' => "Dollar zimbabwéen"],
    ['code' => 'VND', 'name' => "Dong vietnamien"],
    ['code' => 'AMD', 'name' => "Dram arménien"],
    ['code' => 'CVE', 'name' => "Escudo cap-verdien"],
    ['code' => 'HUF', 'name' => "Forint hongrois"],
    ['code' => 'PYG', 'name' => "Guarani paraguayen"],
    ['code' => 'UAH', 'name' => "Hryvnia ukrainienne"],
    ['code' => 'ISK', 'name' => "Króna islandaise"],
    ['code' => 'ZMW', 'name' => "Kwacha zambien"],
    ['code' => 'MMK', 'name' => "Kyat birman"],
    ['code' => 'GEL', 'name' => "Lari géorgien"],
    ['code' => 'ALL', 'name' => "Lek albanais"],
    ['code' => 'HNL', 'name' => "Lempira hondurien"],
    ['code' => 'MDL', 'name' => "Leu moldave"],
    ['code' => 'RON', 'name' => "Leu roumain"],
    ['code' => 'BGN', 'name' => "Lev bulgare"],
    ['code' => 'EGP', 'name' => "Livre égyptienne"],
    ['code' => 'LBP', 'name' => "Livre libanaise"],
    ['code' => 'SDG', 'name' => "Livre soudanaise"],
    ['code' => 'GBP', 'name' => "Livre sterling"],
    ['code' => 'SYP', 'name' => "Livre syrienne"],
    ['code' => 'TRY', 'name' => "Livre turque"],
    ['code' => 'AZN', 'name' => "Manat azerbaïdjanais"],
    ['code' => 'BAM', 'name' => "Mark convertible bosniaque"],
    ['code' => 'MZN', 'name' => "Metical mozambicain"],
    ['code' => 'NGN', 'name' => "Naira nigérian"],
    ['code' => 'ERN', 'name' => "Nakfa érythréenne"],
    ['code' => 'PEN', 'name' => "Nuevo sol péruvien"],
    ['code' => 'TOP', 'name' => "Paʻanga tongien"],
    ['code' => 'MOP', 'name' => "Pataca macanaise"],
    ['code' => 'ARS', 'name' => "Peso argentin"],
    ['code' => 'CLP', 'name' => "Peso chilien"],
    ['code' => 'COP', 'name' => "Peso colombien"],
    ['code' => 'DOP', 'name' => "Peso dominicain"],
    ['code' => 'MXN', 'name' => "Peso mexicain"],
    ['code' => 'PHP', 'name' => "Peso philippin"],
    ['code' => 'UYU', 'name' => "Peso uruguayen"],
    ['code' => 'BWP', 'name' => "Pula botswanais"],
    ['code' => 'GTQ', 'name' => "Quetzal guatémaltèque"],
    ['code' => 'ZAR', 'name' => "Rand sud-africain"],
    ['code' => 'BRL', 'name' => "Real brésilien"],
    ['code' => 'IRR', 'name' => "Rial iranien"],
    ['code' => 'OMR', 'name' => "Rial omanais"],
    ['code' => 'QAR', 'name' => "Rial qatari"],
    ['code' => 'YER', 'name' => "Rial yéménite"],
    ['code' => 'KHR', 'name' => "Riel cambodgien"],
    ['code' => 'MYR', 'name' => "Ringgit malaisien"],
    ['code' => 'SAR', 'name' => "Riyal saoudien"],
    ['code' => 'BYN', 'name' => "Rouble biélorusse"],
    ['code' => 'RUB', 'name' => "Rouble russe"],
    ['code' => 'INR', 'name' => "Roupie indienne"],
    ['code' => 'MUR', 'name' => "Roupie mauricienne"],
    ['code' => 'NPR', 'name' => "Roupie népalaise"],
    ['code' => 'PKR', 'name' => "Roupie pakistanaise"],
    ['code' => 'LKR', 'name' => "Roupie srilankaise"],
    ['code' => 'IDR', 'name' => "Rupiah indonésienne"],
    ['code' => 'ILS', 'name' => "Shekel israélien"],
    ['code' => 'KES', 'name' => "Shilling kényan"],
    ['code' => 'UGX', 'name' => "Shilling ougandais"],
    ['code' => 'SOS', 'name' => "Shilling somalien"],
    ['code' => 'TZS', 'name' => "Shilling tanzanien"],
    ['code' => 'UZS', 'name' => "Som ouzbek"],
    ['code' => 'BDT', 'name' => "Taka bangladais"],
    ['code' => 'KZT', 'name' => "Tenge kazakh"],
    ['code' => 'KRW', 'name' => "Won sud-coréen"],
    ['code' => 'JPY', 'name' => "Yen japonais"],
    ['code' => 'CNY', 'name' => "Yuan chinois"],
    ['code' => 'PLN', 'name' => "Zloty polonais"],
];
