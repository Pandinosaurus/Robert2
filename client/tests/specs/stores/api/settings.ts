import requester from '@/globals/requester';
import apiSettings from '@/stores/api/settings';
import data from '@fixtures/settings';

describe('Settings Api', () => {
    describe('all()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data });
            expect(await apiSettings.all()).toMatchSnapshot();
        });
    });

    describe('update()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data });
            expect(await apiSettings.update({} as any)).toMatchSnapshot();
        });
    });

    describe('reset()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'delete').mockResolvedValue({ data });
            expect(await apiSettings.reset('calendar.public.url')).toMatchSnapshot();
        });
    });
});
