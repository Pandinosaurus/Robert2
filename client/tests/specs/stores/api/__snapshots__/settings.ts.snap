// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Api all() parse the returned data correctly 1`] = `
{
  "billing": {
    "defaultDegressiveRate": 1,
    "defaultTax": 1,
  },
  "calendar": {
    "event": {
      "showBorrower": false,
      "showLocation": true,
    },
    "public": {
      "displayedPeriod": "operation",
      "enabled": true,
      "url": "http://loxya.test/calendar/public/dfe7cd82-52b9-4c9b-aaed-033df210f23b.ics",
    },
  },
  "eventSummary": {
    "customText": {
      "content": null,
      "title": "Contrat",
    },
    "materialDisplayMode": "categories",
    "showDescriptions": false,
    "showLegalNumbers": true,
    "showPictures": false,
    "showReplacementPrices": true,
    "showTags": false,
  },
  "general": {
    "openingHours": [
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 1,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 2,
      },
      {
        "end_time": "12:45:00",
        "start_time": "09:00:00",
        "weekday": 3,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 4,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 5,
      },
      {
        "end_time": "12:00:00",
        "start_time": "10:00:00",
        "weekday": 6,
      },
    ],
  },
  "returnInventory": {
    "mode": "start-empty",
  },
}
`;

exports[`Settings Api reset() parse the returned data correctly 1`] = `
{
  "billing": {
    "defaultDegressiveRate": 1,
    "defaultTax": 1,
  },
  "calendar": {
    "event": {
      "showBorrower": false,
      "showLocation": true,
    },
    "public": {
      "displayedPeriod": "operation",
      "enabled": true,
      "url": "http://loxya.test/calendar/public/dfe7cd82-52b9-4c9b-aaed-033df210f23b.ics",
    },
  },
  "eventSummary": {
    "customText": {
      "content": null,
      "title": "Contrat",
    },
    "materialDisplayMode": "categories",
    "showDescriptions": false,
    "showLegalNumbers": true,
    "showPictures": false,
    "showReplacementPrices": true,
    "showTags": false,
  },
  "general": {
    "openingHours": [
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 1,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 2,
      },
      {
        "end_time": "12:45:00",
        "start_time": "09:00:00",
        "weekday": 3,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 4,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 5,
      },
      {
        "end_time": "12:00:00",
        "start_time": "10:00:00",
        "weekday": 6,
      },
    ],
  },
  "returnInventory": {
    "mode": "start-empty",
  },
}
`;

exports[`Settings Api update() parse the returned data correctly 1`] = `
{
  "billing": {
    "defaultDegressiveRate": 1,
    "defaultTax": 1,
  },
  "calendar": {
    "event": {
      "showBorrower": false,
      "showLocation": true,
    },
    "public": {
      "displayedPeriod": "operation",
      "enabled": true,
      "url": "http://loxya.test/calendar/public/dfe7cd82-52b9-4c9b-aaed-033df210f23b.ics",
    },
  },
  "eventSummary": {
    "customText": {
      "content": null,
      "title": "Contrat",
    },
    "materialDisplayMode": "categories",
    "showDescriptions": false,
    "showLegalNumbers": true,
    "showPictures": false,
    "showReplacementPrices": true,
    "showTags": false,
  },
  "general": {
    "openingHours": [
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 1,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 2,
      },
      {
        "end_time": "12:45:00",
        "start_time": "09:00:00",
        "weekday": 3,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 4,
      },
      {
        "end_time": "17:30:00",
        "start_time": "09:00:00",
        "weekday": 5,
      },
      {
        "end_time": "12:00:00",
        "start_time": "10:00:00",
        "weekday": 6,
      },
    ],
  },
  "returnInventory": {
    "mode": "start-empty",
  },
}
`;
